### PR TITLE
Con 1188 zen desk widget

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs [hunk 1]
```diff
@@ -20,6 +21,44 @@ public static class HtmlHelperExtensions
...
+
+        public static string GetZenDeskSnippetKey(this HtmlHelper html)
+        {
+            return ConfigurationManager.AppSettings["ZenDeskSnippetKey"];
+        }
+
+        public static string GetZenDeskSnippetSectionId(this HtmlHelper html)
+        {
+            return ConfigurationManager.AppSettings["ZenDeskSectionId"];            
+        }
```
Comment (dashton82) [inline src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs]: Are there any tests covering this? Does it matter if these aren't set?

File: src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs [hunk 2]
```diff
@@ -20,6 +21,44 @@ public static class HtmlHelperExtensions
...
+
+        public static MvcHtmlString SetZenDeskLabels(this HtmlHelper html, params string[] labels)
+        {
+            var apiCallString =
+                "<script type=\"text/javascript\">zE('webWidget', 'helpCenter:setSuggestions', { labels: [";
+
+            var first = true;
+            foreach (var label in labels)
+            {
+                if (!string.IsNullOrEmpty(label))
```
Comment (dashton82) [inline src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs]: Should we also check whitespace on this? Not sure if its a scenario we want to cover
Also I'd look at changing it to 
```suggestion
                if (string.IsNullOrWhitepsace(label))
                {
                      continue;
                }
```

File: src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs [hunk 3]
```diff
@@ -20,6 +21,44 @@ public static class HtmlHelperExtensions
...
-        }   
+        }
+
+
+        public static MvcHtmlString SetZenDeskLabels(this HtmlHelper html, params string[] labels)
+        {
+            var apiCallString =
+                "<script type=\"text/javascript\">zE('webWidget', 'helpCenter:setSuggestions', { labels: [";
+
+            var first = true;
```
Comment (dashton82) [inline src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs]:
```suggestion
            var isFirstLabel = true;
```

File: src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs [hunk 4]
```diff
@@ -20,6 +21,44 @@ public static class HtmlHelperExtensions
...
+            foreach (var label in labels)
+            {
+                if (!string.IsNullOrEmpty(label))
+                {
+                    if (!first) apiCallString += ",";
+                    first = false;
```
Comment (dashton82) [inline src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs]: Personal preference but I find it easier to read If statements not on a single line - could also include the setting of the `isFirstLabel` to be inside there

File: src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs [hunk 5]
```diff
@@ -20,6 +21,44 @@ public static class HtmlHelperExtensions
...
+
+                    apiCallString += $"'{ EscapeApostrophes(label) }'";
```
Comment (dashton82) [inline src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs]: Are there likely to be a lot of labels? If so might be better to use StringBuilder

File: src/SFA.DAS.Forecasting.Web.UnitTests/Extensions/HtmlHelperExtensionsTests.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                    first = false;
+
+                    actualOutput += $"'{ EscapeApostrophes(label) }'";
+                }
+            }
+
+            actualOutput += "] });</script>";
+
+            //Act
+            this.actualOutput = HtmlHelperExtensions.SetZenDeskLabels(null, labels).ToString();
```
Comment (dashton82) [inline src/SFA.DAS.Forecasting.Web.UnitTests/Extensions/HtmlHelperExtensionsTests.cs]: can remove `this`

Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.Forecasting.Web.UnitTests/Extensions/HtmlHelperExtensionsTests.cs]: done

File: src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs (hunk 1 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.Forecasting.Web/Extensions/HtmlHelperExtensions.cs]: was thinking initially to add Test for this. After i have the config changes https://github.com/SkillsFundingAgency/das-forecasting-tool/pull/244 , we know the variable been added to the pipeline

**Comment by VasanthaKasirajan3008** [comment]
Please review and let me know

**Pull request merged**